ckanext/hierarchy/plugin.py

```diff
@@ -141,6 +140,17 @@ def before_dataset_search(self, search_params):
             # remove include_children from the filter-list - we have a checkbox
             g.fields_grouped.pop('include_children', None)
 
+        if 'groups' in search_params['fq']:
```

Comment:
I don't like it. This means that sub-group search will be unconditionally applied, whenever group facet is present.
You should do it in a similar manner to the sub-organization search. Ideally, you can integrate your change into `if include_children` clause above.

Reply:
Unfortunately I don't understand how to do it. I've changed the groups search condition using g.controller variable (I've thought it's more convenient and not bind with facet) and passed all tests on all versions of CKAN **except v.2.10** (please see the tests results). Controller condition (in form of toolkit.get_endpoint() which is not compatible with CKAN v.2.7) is used for organizations search in ckanext-hierarchy-search).  Moreover I don't understand how to integrate my changes into `if include_children` clause since after debug I see that this condition is always False and doesn't execute at all.